lab1/app/src/main/java/com/example/labyrinth/game/GameLogic.java

```diff
@@ -1,25 +1,42 @@
-package com.example.labyrinth;
+package com.example.labyrinth.game;
 
 import android.graphics.Point;
 import android.util.Size;
 
+import com.example.labyrinth.Cell;
+import com.example.labyrinth.Direction;
+import com.example.labyrinth.Labyrinth;
+import com.example.labyrinth.LabyrinthGenerator;
+import com.example.labyrinth.Type;
+
 import java.util.function.Consumer;
 
-public class GameLogic {
+public abstract class GameLogic {
```

Comment:
Refactor GameLogic class. Make it abstract class, and some methods with different logic should be overriden in subclasses